backend/auth/auth.go

```diff
@@ -0,0 +1,75 @@
+package auth
+
+import (
+	"fmt"
+	"github.com/MicahParks/keyfunc"
+	"github.com/ZPI-2024-25/KubernetesAccessManager/models"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/joho/godotenv"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+)
+
+var jwks *keyfunc.JWKS
+
+func init() {
+	err := godotenv.Load()
+	jwksURL := os.Getenv("KEYCLOACK_URL")
+	jwks, err = keyfunc.Get(jwksURL, keyfunc.Options{
+		RefreshInterval: time.Hour,
+	})
+	if err != nil {
+		panic(fmt.Sprintf("Failed to create JWKS: %s", err))
+	}
+}
+
+func GetJWTTokenFromHeader(r *http.Request) (string, error) {
+	authHeader := r.Header.Get("Authorization")
+	if authHeader == "" {
+		return "", fmt.Errorf("authorization header missing")
+	}
+
+	parts := strings.Split(authHeader, " ")
+	if len(parts) != 2 || parts[0] != "Bearer" {
+		return "", fmt.Errorf("invalid authorization header format")
+	}
+	return parts[1], nil
+}
+
+func IsTokenValid(tokenStr string) bool {
+	token, err := jwt.Parse(tokenStr, func(token *jwt.Token) (interface{}, error) {
+		return jwks.Keyfunc(token)
+	})
```

Comment:
```suggestion
	token, err := jwt.Parse(tokenStr, jwks.Keyfunc)
```
Can't check, but wouldn't this work?